src/pages/WelcomeProfilePage.js

```diff
@@ -0,0 +1,353 @@
+        this.hideErrorMessage();
+        this.secondaryLoginError = '';
+        this.secondaryLogin = text;
+        this.isLoginValid = text.length === 0 || this.didSecondaryLoginValid(text);
+    }
+
+    displayFormErrorMessage() {
+        if (this.state.showFormError) {
+            return;
+        }
+        this.setState({showFormError: true});
+    }
+
+    hideErrorMessage() {
+        if (!this.state.showFormError && !this.state.requestTimeoutError) {
+            return;
+        }
+        this.setState({showFormError: false, requestTimeoutError: false});
+    }
+
+    /**
+     * Handle screen unmount
+     */
+    handleScreenWillUnmount() {
+        // Reset error message
+        User.clearUserErrorMessage();
+
+        // Don't set current welcome step if current step isn't equal to welcome profile. Could be direct access to the page by url.
+        if (WelcomeAction.getCurrentWelcomeStep() !== CONST.FIRST_TIME_NEW_EXPENSIFY_USER_STEP.WELCOME_PROFILE_SETTING) {
+            return;
+        }
+
+        // Finish current welcome step and move to next step
+        NameValuePair.set(
+            CONST.NVP.FIRST_TIME_NEW_EXPENSIFY_USER_STEP,
+            CONST.FIRST_TIME_NEW_EXPENSIFY_USER_STEP.GLOBAL_CREATE_MENU,
+            ONYXKEYS.NVP_FIRST_TIME_NEW_EXPENSIFY_USER_STEP,
+        );
+    }
+
+    /**
+     * Returns promise of setPersonalDetails request
+     * @return {Promise}
+     */
+    submitPersonalDetails() {
+        return PersonalDetails.setPersonalDetails({
+            firstName: this.firstName.trim(),
+            lastName: this.lastName.trim(),
+        }, false);
+    }
+
+    /**
+     * Returns promise of setSecondaryLogin request
+     * @return {Promise}
+     */
+    submitSecondaryLogin() {
+        const login = this.state.formType === CONST.LOGIN_TYPE.PHONE
+            ? LoginUtil.getPhoneNumberWithoutSpecialChars(this.secondaryLogin)
+            : this.secondaryLogin;
+
+        return User.setSecondaryLogin(login, this.password);
+    }
+
+    /**
+     * Submit form to update personal details
+     */
+    submitForm() {
+        if (!this.didFormValid()) {
+            this.displayFormErrorMessage();
+            return;
+        }
+        if (this.state.waitForResponse) {
+            return;
+        }
+
+        // Set loading state of submit button
+        this.setState({waitForResponse: true, requestTimeoutError: false});
+
+        let submitPersonalDetailsPromise;
+        let submitSecondaryLoginPromise;
+        if (this.firstName.trim().length + this.lastName.trim().length > 0) {
+            submitPersonalDetailsPromise = this.submitPersonalDetails();
+        }
+
+        if (this.secondaryLogin.trim().length > 0) {
+            submitSecondaryLoginPromise = this.submitSecondaryLogin();
+        }
+
+        // Reject promise request after soome ms
+        const rejectAfterDelay = ms => new Promise((resolve, reject) => {
+            setTimeout(reject, ms, 'timeout');
```

Comment:
Can you please help me understand why the request will be rejected after 8000 ms?
I think we should remove this

Reply:
I'm sorry, could you add some steps so that it's clear why this is an issue? Thanks

Reply:
In welcome profile page, please test the submit of the form. Fill fill first name, last name, phone number = + 123. 
Press `get started` repeatedly (each time `get started` appear not on loading state of the button), if you don't reach message `network request taking too long` error then maybe we don't need to add that code, maybe it is just my bad connection.
The code above is needed for request that is taking too long or connection is keep on pending. I don't know that is the cause, but I test it and it is exist. It keep going on waiting for response of the server for long time. We need to display error message from the server in the form for phone validation, so I made it to wait for server response.

Reply:
@K4tsuki here are some suggestions 

- Don't block on personal details. Let it be async, we don't care if it fails or errors out.
- Use `ONYXKEYS.ACCOUNT` to get loading state and error message. (see setPasswordPage for reference)

Doing this will greatly simplify the code, and we can then remove the entire promise logic from this file.
Please let me know if you have any questions, thanks!

Reply:
@rushatgabhane just for confirmation, If there is an error, we will just display `growl message` ? Then proceed to next step to display global create menu? Right?

Reply:
@K4tsuki if there's an error for invalid phone number. We can just show it under the input

<img width="278" alt="Screen Shot 2022-03-10 at 2 27 54 AM" src="https://user-images.githubusercontent.com/29673073/157556682-70a8cbc9-f348-4ca3-b5b3-9f328e1a1a21.png">

Reply:
@rushatgabhane Ok, then that is what I did. 
For phone validation, we need server response so we wait for it. 
We don't have functions to validate the phone number.  I mean this error message: `I couldn't validate the phone number, please try again with the country code (e.g. +15005550006).` It is from server and we don't have functions to validate it that is why I wait for the response and display the error message.

I think there is a misunderstanding:

> Don't block on personal details. Let it be async, we don't care if it fails or errors out.

So then I am not blocking, I just  show loading state of the button. I am not blocking anything/freezing ui or anything here. What I mean with blocking is `show loading state of the button while wait for server response`

Reply:
> For phone validation, we need server response so we wait for it.
We don't have functions to validate the phone number. I mean this error message: I couldn't validate the phone number, please try again with the country code (e.g. +15005550006). It is from server and we don't have functions to validate it that is why I wait for the response and display the error message.

Sounds great! After https://github.com/Expensify/App/pull/8019/files is merged,
for this particular case you can check if `jsonCode` is `402` and show common.error.phoneNumber

<hr/>

> I think there is a misunderstanding: Don't block on personal details. Let it be async, we don't care if it fails or errors out. So then I am not blocking, I just show loading state of the button. I am not blocking anything/freezing ui or anything here. It is async operation, I am using promise. What I mean with blocking is show loading state of the button while wait for server response.

Oh okay, thanks for clearing that. What I mean to say is - the button shouldn't be loading based on `setPersonalDetails()` (optimistic action)

The button should load only when waiting for a response from `setSecondaryLogin()` (pessimistic action)

<hr/>

> I am limiting the wait for response for 8 seconds then show request taking too long message. There are many factor that make request pending or taking too long to complete. Just to be safe here, I am limiting the wait for 8 seconds, and we could lower the wait time.

I agree, but we haven't standardized on how to handle this particular case. Letting the loader spin until the request is complete is good for now. I hope this makes sense, thanks!

Reply:
Okay